docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,76 @@
+# Architecture
+
+This library is under development, and it's primarily designed to fulfil needs of the
+open telemetry exporter and therefore some functionality that would be useful in other
+scenarios is not implemented.
+
+## Idea
+
+Design is heavily influenced by peculiarities of the upstream and downstream systems
+about which I know very little.
+
+### Downstream
+
+The downstream system is DataSet REST API and its API call [addEvents](https://app.scalyr.com/help/api#addEvents).
+It has the following properties:
+
+* `D1` - only one request for each session could be made
```

Comment:
Should probably clarify it that this means that only one request per session can be active at one time (aka no parallel questions per session).

Could also add that this - network latency and RTT (round trip time) can have large affect on the throughput.

Reply:
updated